src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -123,13 +123,23 @@ const DateTimeBlock : React.FC<IProps> = ({
   const convertMinutes = (date?.getMinutes()).toString();
   const [displayHours, setDisplayHours] = useState<string>(getFormattedTime(convertHours));
   const [displayMinutes, setDisplayMinutes] = useState<string>(getFormattedTime(convertMinutes));
+  const [isKeyboardEvent, setIsKeyboardEvent] = useState(false);
+
+  const handleKeyDown = useCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
+    const key = event.key;
+    if (key === "ArrowUp" || key === "ArrowDown") {
+      setIsKeyboardEvent(false);
+    } else {
+      setIsKeyboardEvent(true);
+    }
```

Comment:
Here, Every event is an keyboard event so I think we need to reconsider name of this state something more explanatory.

Reply:
Done